include/pybind11/attr.h

```diff
@@ -377,7 +377,7 @@ template <> struct process_attribute<is_new_style_constructor> : process_attribu
 };
 
 inline void process_kw_only_arg(const arg &a, function_record *r) {
-    if (!a.name || strlen(a.name) == 0)
+    if (!a.name || a.name[0] == '\0')
```

Comment:
Would this be cleaner this way?
```suggestion
    if (!a.name || !a.name[0])
```

Maybe the first way reads better, though. Just a thought.

Reply:
I am personally against implicit boolean conversions for readability reasons.

Reply:
Yeah, I don't think I like it that way either. The current way clearly indicates what it's doing.

Reply:
Both ways are equivalent. However, I also want to make the intention more explicit.